File: Aditya Vardhan Singh/Other Problems/Day 13/160 Intersection of Two Linked Lists/Python/160 Intersection of Two Linked Lists.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution:
    def getIntersectionNode(self, headA: ListNode, headB: ListNode) -> Optional[ListNode]:
        st = set()
        temp = headA
        while temp:
            st.add(temp)
            temp = temp.next
        temp = headB
        while temp:
            if temp in st:
                return temp
            temp = temp.next
        return None

File: Aditya Vardhan Singh/Other Problems/Day 13/160 Intersection of Two Linked Lists/C++/160 Intersection of Two Linked Lists.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode(int x) : val(x), next(NULL) {}
 * };
 */
#include <unordered_set>
class Solution {
public:
    ListNode *getIntersectionNode(ListNode *headA, ListNode *headB) {
        unordered_set<ListNode *> st;
        ListNode * temp = headA;
        while (temp) {
            st.insert(temp);
            temp = temp->next;
        }
        temp = headB;
        while (temp) {
            if (st.find(temp) != st.end())
                return temp;
            temp = temp->next;
        }
        return nullptr;
    }
};